clarity/src/vm/contexts.rs

```diff
@@ -441,10 +444,7 @@ impl AssetMap {
         asset_identifier: &AssetIdentifier,
     ) -> Option<u128> {
         match self.token_map.get(principal) {
```

Comment:
This could be further simplified I think:

```rust
        self.token_map.get(principal)
            .map(|assets| assets.get(asset_identifier).copied())
```